src/views/Deposit/components/EnterAmount/index.tsx

```diff
@@ -22,87 +21,65 @@ import {
+import { API as OnboardAPI } from 'libs/types';
 
-// Utils
-import { approve, bnToDec, decToBn, getAllowance, getNotification, stake } from 'utils';
+import { stake } from 'utils';
+
+import useApproval from 'hooks/useApproval';
 
 interface IEnterAmount {
   farmSelected: boolean;
+  readyToTransact: (onboard: OnboardAPI | null, provider: Provider | null) => Promise<boolean>;
   setFarmSelected: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
-export const EnterAmount: React.FC<IEnterAmount> = ({ farmSelected, setFarmSelected }) => {
+export const EnterAmount: React.FC<IEnterAmount> = ({ farmSelected, readyToTransact, setFarmSelected }) => {
   const { wpoktBalance } = React.useContext(BalanceContext);
-  const [isApproved, setIsApproved] = React.useState<boolean>(true);
   const [isDisabled, setIsDisabled] = React.useState<boolean>(true);
   const [wpoktInputValue, setWpoktInputValue] = React.useState<string>('');
-  const { address, onboard, provider, notify, signer } = React.useContext(Web3Context);
+  const { address, onboard, provider, signer } = React.useContext(Web3Context);
+  const { isApproved, isApproving, onApprove } = useApproval();
 
-  // Should probably a separated hook
-  const useApproval = async () => {
-    if (address && provider && !isDisabled && farmSelected) {
-      const allowance = await getAllowance(address, TOKEN_GEYSER_ADDRESS, WPOKT_ADDRESS, provider);
-      if (+decToBn(+wpoktInputValue) > +allowance) {
-        setIsApproved(true);
-      } else {
-        setIsApproved(false);
-      }
-    }
-  };
-  useApproval();
-
-  // Effects
   React.useEffect(() => {
     if (wpoktInputValue === '' || wpoktInputValue === '0' || !farmSelected) {
       setIsDisabled(true);
     } else {
       setIsDisabled(false);
     }
-  }, [wpoktInputValue, farmSelected]);
+  }, [address, farmSelected, wpoktInputValue]);
+
+  // This is a placeholder, which will eventually launch the transaction status modal
+  React.useEffect(() => {
+    if (isApproving) {
+      console.log('Approving...');
+    } else if (isApproved) {
+      console.log('Approved');
+    }
+  }, [isApproving, isApproved]);
 
-  // Handlers
   const onDeposit = async () => {
-    onboard?.walletCheck();
-    if (address && provider && !isDisabled && farmSelected) {
-      if (isApproved && signer) {
-        const response = await approve(
-          decToBn(+wpoktInputValue).toString(),
-          TOKEN_GEYSER_ADDRESS,
-          WPOKT_ADDRESS,
-          signer,
-        );
-        if (typeof response === 'boolean') {
-          console.log(response);
-        } else {
-          if (notify) {
-            getNotification(notify, response);
-          }
-        }
+    readyToTransact(onboard, provider);
+    if (address && signer && !isDisabled && farmSelected) {
+      if (isApproved) {
+        const dummyToken = new TokenAmount('1', 0);
+        const convertedAmount = dummyToken.convert(wpoktInputValue, 18).value;
```

Comment:
I don't quite get these changes with removing `bignumber.js`—how are we handling user input now for decimals? Are we just converting it to a string and displaying that on the "front", while keeping the bignumber on the "back"? `TokenAmount` should only be used for formatting purposes or converting from rates, so I'm a bit confused. 😅

Reply:
I actually was more or less copying these tests to handle decimals: https://github.com/aragon/token-amount/blob/master/src/TokenAmount.test.ts#L26

Bit of a cheat maybe, but it seemed that you could use the convert method to change decimal inputs to "non-formatted" values.

Reply:
Honestly, wasn't sure of a good way to convert decimal inputs to "non-formatted" values if we aren't using `bignumber.js`. But this seems to weirdly do the trick. Do you think this will still cause precision issues, or does it just look confusing haha?

Reply:
Yeah, this is definitely confusing haha... I can see this works for converting from decimals to amounts, but if so, let's write a bunch of tests for this to make sure this won't either:
1 - Give wrong values for big or small amounts (a lot of decimals, or a huge amount being given to the input)
2 - be inaccurate at any point, no matter the amount

If we wanna switch to a battle-tested implementation, I'd take a look at this hook—we can probably strip out a bunch of stuff from this and use it, even though it would mean using Ethers's `BigNumber` type (which is fine because we already have it).

Reply:
Sorry if I seem very bikesheddy on this part of the app... But really, after we get this form right, everything else is pretty much downhill :)

Reply:
Ah, no worries, no worries. So I guess the crux of this hook is really just using [`EthersUtils.parseUnits()`](https://github.com/aragon/upgrade/blob/67fdd07f411f210daf1fb9e23450bcac8461cd52/src/utils/math-utils.ts#L15), right? If so, I'll just pop that in in place of what I have!

Reply:
Alright, decided to go the battle-tested route: 8a6f067dfaa8a6ae95bce9a65c198df33172dcb4